agent/consul/multilimiter/multilimiter.go

```diff
@@ -88,13 +100,35 @@ func NewMultiLimiter(c Config) *MultiLimiter {
 	if c.ReconcileCheckInterval == 0 {
 		c.ReconcileCheckLimit = 1 * time.Second
 	}
-	m := &MultiLimiter{limiters: &limiters, defaultConfig: &config, limitersConfigs: &configs}
+	chLimiter := make(chan *limiterWithKey, 100)
+	m := &MultiLimiter{limiters: &limiters, defaultConfig: &config, limitersConfigs: &configs, limiterCh: chLimiter, configsLock: sync.Mutex{}}
+
 	return m
 }
 
 // Run the cleanup routine to remove old entries of Limiters based on ReconcileCheckLimit and ReconcileCheckInterval.
 func (m *MultiLimiter) Run(ctx context.Context) {
+	isRunning := m.isRunning.Load()
+	if isRunning {
+		return
+	}
+	m.isRunning.Store(true)
```

Comment:
This is thread-unsafe because `isRunning` could change after the `Load` — so we need to do an atomic CAS:

```suggestion
	if swapped := m.isRunning.CompareAndSwap(false, true); !swapped {
		return
	}
```

Alternatively, if we don't need to re-run once it has stopped, the whole thing could be replaced with a `sync.Once` 1️⃣